internal/app/machined/pkg/controllers/k8s/kubelet_spec_test.go

```diff
@@ -184,3 +241,102 @@ func (suite *KubeletSpecSuite) TearDownTest() {
 func TestKubeletSpecSuite(t *testing.T) {
 	suite.Run(t, new(KubeletSpecSuite))
 }
+
+func TestNewKubeletConfigurationFail(t *testing.T) {
+	for _, tt := range []struct {
+		name        string
+		extraConfig map[string]interface{}
+		expectedErr string
+	}{
+		{
+			name: "wrong fields",
+			extraConfig: map[string]interface{}{
+				"API":  "v1",
+				"foo":  "bar",
+				"Port": "xyz",
+			},
+			expectedErr: "error unmarshalling extra kubelet configuration: strict decoding error: unknown field \"API\", unknown field \"Port\", unknown field \"foo\"",
+		},
+		{
+			name: "wrong field type",
+			extraConfig: map[string]interface{}{
+				"oomScoreAdj": "v1",
+			},
+			expectedErr: "error unmarshalling extra kubelet configuration: unrecognized type: int32",
```

Comment:
shouldn't it say `string`  instead?

Reply:
it's not us in this case, it's the way Kubernetes converter unmarshaller the error, and we have no control over it

Reply:
okay, it was confusing though :sweat_smile: